internal/controller/drcluster_controller.go

```diff
@@ -615,6 +624,80 @@ func (u *drclusterInstance) finalizerRemove() error {
 		Update(u.ctx, u.client)
 }
 
+func (u *drclusterInstance) ensureDRClusterConfig() error {
+	drcConfig, err := u.generateDRClusterConfig()
+	if err != nil {
+		return err
+	}
+
+	if err := u.mwUtil.CreateOrUpdateDRCConfigManifestWork(u.object.Name, *drcConfig); err != nil {
+		return fmt.Errorf("failed to create or update DRClusterConfig manifest on cluster %s (%w)",
+			u.object.GetName(), err)
+	}
+
+	if !u.mwUtil.IsManifestApplied(u.object.Name, util.MWTypeDRCConfig) {
+		return fmt.Errorf("DRClusterConfig is not applied to cluster (%s)", u.object.Name)
+	}
+
+	return nil
+}
+
+func (u *drclusterInstance) generateDRClusterConfig() (*ramen.DRClusterConfig, error) {
+	mc, err := util.NewManagedClusterInstance(u.ctx, u.client, u.object.GetName())
+	if err != nil {
+		return nil, err
+	}
+
+	clusterID, err := mc.ClusterID()
+	if err != nil {
+		return nil, err
+	}
+
+	drcConfig := ramen.DRClusterConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "DRClusterConfig",
+			APIVersion: "ramendr.openshift.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: u.object.GetName(),
+		},
+		Spec: ramen.DRClusterConfigSpec{
+			ClusterID: clusterID,
+		},
+	}
+
+	drpolicies, err := util.GetAllDRPolicies(u.ctx, u.reconciler.APIReader)
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensure that schedules are not duplicated by, storing them in "added" to avoid adding a duplicate schedule from
+	// another DRPolicy
+	added := map[string]bool{}
+
+	for idx := range drpolicies.Items {
```

Comment:
Is it possible, that policy is marked for deletion? Do we need to handle it anyway?

Reply:
Interestingly there is a test case for this that passes, though based on the comment I think it is possible that we may retain a stale schedule till a future DRCluster reconcile (due to any reason) is initiated. Will fix this by skipping deleted DRPolicies in the loop.

The parent PR that this is a part of is merged, hence will address it as a separate PR.